### PR TITLE
added history, view count, like count functionality

### PR DESCRIPTION
**Pull request by JantuDeb**
# Video History
- add(on viewed)  and remove videos from history
- like and dislike videos.
- View count 
image
image

### CODE REVIEW

File: src/components/shared/video-card/VideoCard.css [hunk 1]
```diff
@@ -40,3 +40,19 @@
...
+
+.history{
+  position: relative;
+}
+.history-close-icon{
+  width: 1.8rem;
+  height: 1.8rem;
+  position: absolute;
+
+  top: 0.5rem;
```
Comment (ShreyashMahajan) [inline src/components/shared/video-card/VideoCard.css:53]: try to use px for less than 1rem

File: src/context/history/HistoryContext.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+const HistoryVideoContext = createContext([]);
+const HistoryVideoProvider = ({ children }) => {
+  const [historyVideos, historyDispatch] = useReducer(historyReducer, []);
+
+  /**
+   * Get historyvideos from api
+   */
+  const getHistoryVideos = async () => {
+    try {
+      const { data } = await axioxPrivate.get("/user/history");
```
Comment (sushil-mod) [inline src/context/history/HistoryContext.jsx:19]: I liked how you created axios instance for private

**Review by ShreyashMahajan — APPROVED** [review]
Looks good to merge

File: src/context/history/HistoryContext.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      if (data.success) {
+        const historyVideos = data.histories.map((history) => history.video);
+        historyDispatch({
+          type: GET_HISTORY_VIDEOS,
+          payload: historyVideos,
+        });
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (sushil-mod) [inline src/context/history/HistoryContext.jsx:28]: if possible you can remove console

**Review by sushil-mod — APPROVED** [review]
UI looking awesome ,nice and clean code . I learned how you created reducer for different functionality .nice and clean code

**Pull request merged**